### PR TITLE
Adds caching of pods and gems for Travis

### PR DESCRIPTION
**Pull request by koke**
Let's see if this works
https://blog.travis-ci.com/2016-05-03-caches-are-coming-to-everyone

If the travis build is green and the [caches](https://travis-ci.org/wordpress-mobile/WordPress-iOS/caches) are properly generated, this should be enough.

Needs review: @SergioEstevao

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -3,11 +3,11 @@ language: objective-c
...
 xcode_scheme: WordPress
 xcode_sdk: iphonesimulator
+cache:
+- bundler
+- cocoapods
 notifications:
   email: true
-install:
-- gem install xcpretty-travis-formatter
-- pod install
```
Comment (SergioEstevao) [inline .travis.yml]: Don't we need to run pod install anymore?

Comment (koke) [inline .travis.yml]: Travis runs `pod install` by default, as long as there's a `Podfile` and you don't override the `install` step

https://docs.travis-ci.com/user/languages/objective-c#Dependency-Management

**Comment by koke** [comment]
Looks like cache isn't being stored, reached out to travis on Twitter for support

https://twitter.com/koke/status/728271201899061249

PR should be good to go, but we won't be able to verify it works until that's fixed.

**Comment by SergioEstevao** [comment]
:shipit:

**Pull request merged**